### PR TITLE
 SECURITY-36 3.10.10

### PR DESCRIPTION
**Pull request by bio-boris**
# Description of PR purpose/changes

- Update clients to work with mongo7 by updating pymongo and mongoengine
-  Fix linting issues from flake8 and black
- Update outdated libraries that can no longer be pip installed
- The latest Dockerfile build contains conda which installs python 3.12.4 so these changes will pin python to 3.10 and save us headache in the future
- Speed up tests by moving docker build step out
- Create matrix of tests to run against mongo3.6 and mongo7

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -147,7 +147,7 @@ services:
       ZOOKEEPER_SERVERS: zookeeper:12888:13888
 
   mongodb:
-    image: mongo:3.2
+    image: mongo:7.0
```
Comment (Xiangs18) [inline docker-compose.yml:150]: We only test against mongo3 in this PR.

File: requirements.txt [hunk 2]
```diff
@@ -7,17 +7,16 @@ configparser==5.0.2
...
+htcondor==23.8.1
 Jinja2==3.0.1
 JSONRPCBase==0.2.0
 mock==4.0.3
 maps==5.1.1
-mongoengine==0.23.1
+mongoengine==0.28.2
 psutil==5.8.0
-pymongo==3.12.0
+pymongo==4.8.0
```
Comment (Xiangs18) [inline requirements.txt:19]: revert back pymongo version

File: docker-compose.yml (hunk 1 above)
Comment (bio-boris) [reply docker-compose.yml:150]: I thought we agreed to add tests against both in the next PR?

Comment (Xiangs18) [reply docker-compose.yml]: 👍

File: requirements.txt (hunk 2 above)
Comment (Xiangs18) [reply requirements.txt:19]: 👍

**Comment by bio-boris** [comment]
@Xiangs18 Kafka seems to stay up when I try it, but zookeeper crashes. It looks like the tests still pass (But throw warnings in the logs when they are removed the docker-compose file as per  https://github.com/kbase/execution_engine2/pull/485/ )

**Comment by Xiangs18** [comment]
Screenshot 2024-08-06 at 1 02 47 PM

**Comment by bio-boris** [comment]
We discussed this oob
* We deployed this to CI
* Ran a job
* Checked job logs
* Checked the job browser

And things are working

**Review by Xiangs18 — APPROVED** [review]
LGTM

**Pull request merged**